### ISSUE TITLE
onion_key file not found if path includes symlink (tor connection, `dualstack` mode)

### ISSUE THREAD

**Issue by pinheadmz** [OpenBazaar/openbazaar-go#826]
Because OB builds a large database on disk of IPFS blocks I used a symlink to move the OB data directory to an external SSD USB drive:

`lrwxrwxrwx  1 pi   pi         24 Dec 17 23:33 .openbazaar2.0 -> /home/pi/usb/openbazaar/`

Every OB function worked totally normally and great until I tried to connect OB to tor. After trying all kinds of different configuration options the last error message I couldn't solve was this in the OB log:

`12:59:31.965 [Execute] [ERROR] failed to listen on any addresses: [missing onion service key material for <MY-ONION-PUBLIC-KEY>]`

To fix, I rebuilt the default data directory at `~/.openbazaar2.0`, copied all the flat files including my `onion_key` file, and just used symlinks to redirect the big data directories (`blocks`, `logs`, `outbox`...) back to the external SSD drive. Success! For the first time I am seeing the onion address in the OB log on startup:
```
15:41:28.630 [printSwarmAddrs] [INFO] Swarm listening on /onion/<MY-ONION-PUBLIC-KEY>:4003
```

**Comment by cpacia** [comment]
I think you said you got this fixed and it was a symlink issue right?

**Comment by pinheadmz** [comment]
Yeah just added to Github in case some one else has the issue or if there's a way to fix it. The symlink wasn't broken or anything it just seemed like the OB server couldn't follow it when looking for the onion key.

**Comment by pinheadmz** [comment]
Ok so my solution above really hosed everything else on my server. 
https://gateway.ob1.io/ipns/QmeSyTRaNZMD8ajcfbhC8eYibWgnSZtSGUp3Vn59bCnPWC
had a message that said `"Can't follow symlinks"` So I put my data directory back to how it was before: the ENTIRE directory symlinked to my usb drive:

`lrwxrwxrwx  1 pi   pi         23 Jan 18 15:26 .openbazaar2.0 -> /home/pi/usb/openbazaar `

Which means that the server CAN'T find the onion key for whatever reason, and doesn't connect to tor. This is still an issue I'd like help to fix. I really need to keep the blocks and datastore on an external drive, but also run in dual-stack mode!

**Comment by placer14** [comment]
Opened a PR against the https://github.com/OpenBazaar/go-onion-transport repo which will be pulled upstream once the `gx publish` voodoo is performed by @cpacia
